### PR TITLE
fix(gradle): invoke gradlew directly, not with /bin/sh (#4630)

### LINKED ISSUE #4630: getGradleCommandLine should not use `sh`
**What Renovate type are you using?**
Self-hosted, renovate/renovate:latest.

**Describe the bug**
https://github.com/renovatebot/renovate/blob/24a9e290fdc5da476b3d22652fa111a52f308b63/lib/manager/gradle/index.ts#L151-L153
Renovate invokes `sh` but inside of the docker image, `sh` is symlinked to `/bin/dash`, which finds invalid syntax in `gradlew`:
```
$ sh gradlew
gradlew: 158: gradlew: Syntax error: "(" unexpected
$ cat -n gradlew | grep -C3 158
   155	fi
   156
   157	# Split up the JVM_OPTS And GRADLE_OPTS values into an array, following the shell quoting and substitution rules
   158	function splitJvmOpts() {
   159	    JVM_OPTS=("$@")
   160	}
   161	eval splitJvmOpts $DEFAULT_JVM_OPTS $JAVA_OPTS $GRADLE_OPTS
```
This bash syntax is not valid in dash.  `gradlew` ought to be treated as executable and executed directly, I think.

**Did you see anything helpful in debug logs?**
```
 WARN: Gradle command sh gradlew --init-script renovate-plugin.gradle renovate failed. Exit code: 2. (repository=foo/bar)
       "err": {
         "killed": false,
         "code": 2,
         "signal": null,
         "cmd": "sh gradlew --init-script renovate-plugin.gradle renovate",
         "stdout": "",
         "stderr": "gradlew: 158: gradlew: Syntax error: \"(\" unexpected\n",
         "message": "Command failed: sh gradlew --init-script renovate-plugin.gradle renovate\ngradlew: 158: gradlew: Syntax error: \"(\" unexpected\n",
         "stack": "Error: Command failed: sh gradlew --init-script renovate-plugin.gradle renovate\ngradlew: 158: gradlew: Syntax error: \"(\" unexpected\n\n    at ChildProcess.exithandler (child_process.js:294:12)\n    at ChildProcess.emit (events.js:198:13)\n    at ChildProcess.EventEmitter.emit (domain.js:448:20)\n    at maybeClose (internal/child_process.js:982:16)\n    at Socket.stream.socket.on (internal/child_process.js:389:11)\n    at Socket.emit (events.js:198:13)\n    at Socket.EventEmitter.emit (domain.js:448:20)\n    at Pipe._handle.close (net.js:606:12)"
       }
```

**To Reproduce**

1. Generate a Gradle project with Gradle 3.4.1 (I don't know that a version this old is necessary but it's the version that generated this project's `gradlew`).
2. Try to use renovate/renovate:latest on the resulting project.

### PR DESCRIPTION
**Pull request by isker**
Closes #4630.  In addition to the automated test, I tested this manually with the project that I described in that issue.

### CODE REVIEW

**Review by ikesyo — DISMISSED** [review]
LGTM, this is a usual way to use gradlew 👍

**Comment by JamieMagee** [comment]
The current method prevents us from having the execute bit set. I know that 99% of `gradlew` should have this set, but I've seen cases where it's not.

If it's a quick fix could you add a check for that here? Otherwise I'll open a separate issue to track it.

**Comment by isker** [comment]
@JamieMagee right, okay.  I can do that.  And you think we should continue to use `sh` in the case that it's not executable?  I don't know if we can do anything smarter than that.

**Comment by JamieMagee** [comment]
@isker Yeah, I think that's about the best we can do.

**Comment by isker** [comment]
@JamieMagee @ikesyo how's this?

File: lib/manager/gradle/index.ts [hunk 1]
```diff
@@ -156,4 +161,13 @@ async function getGradleCommandLine(
   return cmd + ' ' + GRADLE_DEPENDENCY_REPORT_OPTIONS;
 }
 
+async function canExecute(path) {
```
Comment (ikesyo) [inline lib/manager/gradle/index.ts]:
```suggestion
async function canExecute(path: string): boolean {
```

Comment (isker) [reply lib/manager/gradle/index.ts]: Hmm, given that the function is not exported, I'm not sure why we can't let the compiler infer the types, but this is fine too.

**Review by JamieMagee — APPROVED** [review]

**Pull request merged**